app/controllers/vacancies_controller.rb

```diff
@@ -75,4 +79,13 @@ def smoke_test?
   def audit_row
     @jobs_search_form.to_hash.merge(total_count: @vacancies_search.vacancies.raw_answer["nbHits"])
   end
+
+  def authenticate_for_map
+    return unless params[:location]&.include?(SEARCH_AFFIX_FOR_MAP)
+
+    params[:location].gsub!(SEARCH_AFFIX_FOR_MAP, "")
+    @authenticated_for_map = authenticate_or_request_with_http_basic do |user, password|
+      user == ENV.fetch("MAP_USERNAME") && password == ENV.fetch("MAP_PASSWORD")
```

Comment:
Could we re-use the Sidekiq auth username/password so we don't need to have another separate set of credentials?

Reply:
On second thought, is this really so sensitive we need to authenticate at all? IMHO just the params bit would be good enough...